windows/test-app.ts

```diff
@@ -0,0 +1,300 @@
+    replacements: Replacements,
+): string {
+    let content = fs.readFileSync(srcPath, 'utf8');
+
+    Object.keys(replacements).forEach(regex => {
+        content = content.replace(new RegExp(regex, 'g'), replacements[regex]);
+    });
+
+    return content;
+}
+
+// Binary files, don't process these (avoid decoding as utf8)
+const binaryExtensions = ['.png', '.jar', '.keystore'];
+
+/**
+ * Copy a file to given destination, replacing parts of its contents.
+ * @param srcPath Path to a file to be copied.
+ * @param destPath Destination path.
+ * @param replacements: e.g. {'TextToBeReplaced': 'Replacement'}
+ */
+async function copyAndReplace(
+    srcPath: string,
+    destPath: string,
+    relativeDestPath: string,
+    replacements: Replacements = {}
+) {
+    const fullDestPath = path.join(destPath, relativeDestPath);
+    if (fs.lstatSync(srcPath).isDirectory()) {
+        if (!fs.existsSync(fullDestPath)) {
+            fs.mkdirSync(fullDestPath);
+        }
+        return;
+    }
+
+    const extension = path.extname(srcPath);
+    if (binaryExtensions.indexOf(extension) !== -1) {
+        // Binary file
+        copyBinaryFile(srcPath, fullDestPath, err => {
+            if (err) {
+                throw err;
+            }
+        });
+    } else {
+        // Text file
+        const srcPermissions = fs.statSync(srcPath).mode;
+        let content = resolveContents(srcPath, replacements);
+
+        fs.writeFileSync(fullDestPath, content, {
+            encoding: 'utf8',
+            mode: srcPermissions,
+        });
+    }
+}
+
+/**
+ * Same as 'cp' on Unix. Don't do any replacements.
+ */
+function copyBinaryFile(
+    srcPath: string,
+    destPath: string,
+    cb: (err?: Error) => void,
+) {
+    let cbCalled = false;
+    const srcPermissions = fs.statSync(srcPath).mode;
+    const readStream = fs.createReadStream(srcPath);
+    readStream.on('error', err => {
+        done(err);
+    });
+    const writeStream = fs.createWriteStream(destPath, {
+        mode: srcPermissions,
+    });
+    writeStream.on('error', err => {
+        done(err);
+    });
+    writeStream.on('close', () => {
+        done();
+    });
+    readStream.pipe(writeStream);
+    function done(err?: Error) {
+        if (!cbCalled) {
+            cb(err);
+            cbCalled = true;
+        }
+    }
+}
+
+export function createDir(destPath: string) {
+    if (!fs.existsSync(destPath)) {
+        fs.mkdirSync(destPath);
+    }
+}
+
+export async function copyAndReplaceAll(
+    srcPath: string,
+    destPath: string,
+    relativeDestDir: string,
+    replacements: Replacements = {}
+) {
+    for (const absoluteSrcFilePath of walk(srcPath)) {
+        const filename = path.relative(srcPath, absoluteSrcFilePath);
+        const relativeDestPath = path.join(relativeDestDir, filename);
+        await copyAndReplace(
+            absoluteSrcFilePath,
+            destPath,
+            relativeDestPath,
+            replacements
+        );
+    }
+}
```

Comment:
We should allow `copyAndReplace` to be run in parallel or fire-and-forgot:
```suggestion
export function copyAndReplaceAll(
    srcPath: string,
    destPath: string,
    relativeDestDir: string,
    replacements: Replacements = {}
): Promise<void> {
    return Promise.all(walk(srcPath).map((absoluteSrcFilePath) => {
        const filename = path.relative(srcPath, absoluteSrcFilePath);
        const relativeDestPath = path.join(relativeDestDir, filename);
        return copyAndReplace(
            absoluteSrcFilePath,
            destPath,
            relativeDestPath,
            replacements
        );
    }));
}
```